_posts/2022-03-22-publiccode-yaml-analyse.md

```diff
@@ -0,0 +1,106 @@
+
+**Programmiersprachen**<br />
+Die Topliste der verwendeten Programmiersprachen entspricht weitestgehend den diversen im Internet kursierenden Listen der meistgenutzten Programmiersprachen weltweit. Java liegt mit 66 Repositories bzw. 22% höher als erwartet. Aber wenn man genauer hinschaut, stellt man fest, dass z.B. die ["Regione Marche"](https://github.com/regione-marche) sehr viele Java-Repositories veröffentlicht, die allesamt ein und der gleichen E-Procurement-Lösung zuzuordnen sind. Dass eine Software, die z.B. in Microservice- oder Plugin-Architektur erstellt wurde und daher über viele Einzelrepositories verteilt ist, in dieser Liste überpropoertional einfließt, ist ein Gesichtspunkt, der in dieser Analyse nicht näher berücksichtigt wird, und generell die Ergebnisse verfälschen kann.
+
+## Anwendungstypen
+
+Was könnte man sich sonst noch anschauen? z.B. welchen Anwendungstypen und welchen Themen sind die geförderten Software-Repositories zuzuordnen?
+
+ <iframe width="100%" height="410" scrolling="no" src="https://tomsrocket.github.io/github-publiccode/public/analysis-3.html"></iframe>
+Quelle: [https://tomsrocket.github.io/github-publiccode/public/analysis.html](https://tomsrocket.github.io/github-publiccode/public/analysis.html)
+<br />
+
+ <iframe width="100%" height="410" scrolling="no" src="https://tomsrocket.github.io/github-publiccode/public/analysis-4.html"></iframe>
+Quelle: [https://tomsrocket.github.io/github-publiccode/public/analysis.html](https://tomsrocket.github.io/github-publiccode/public/analysis.html)
+<br />
+
+## Fazit
+
+Dass öffentlich geförderter Code als Open Source bereitgestellt werden muss, ist eine sehr gute Vorgabe. Man muss bedenken, dass es sich hier um Code handelt, der üblicherweise mit Steuergeldern bezahlt wurde, und somit sollte auch der Quellcode als digitales Allgemeingut wieder an die Allgemeinheit zurückfließen.
+
+Das Auffinden solchen Codes wäre normalerweise in den Weiten des Internets fast unmöglich. Dass diese Code-Repositories eine "publiccode.yml"-Datei enthalten müssen, ist zwar eine gute Idee, funktioniert aber nur eingeschränkt. In diesem Fall wurden nur Repositories betrachtet, die bei Github verwaltet werden und das dürfte nur auf einen Bruchteil der öffentlich geförderten Repositories zutreffen.
+
+## Ausblick
+
+In Frankreich gibt es eine zentrale Liste allen öffentlich geförderten Codes. Diese Liste könnte in einem nächsten Schritt betrachtet werden.
+
+In Deutschland gibt es Bestrebungen, ebenfalls die "publiccode.yml" zu nutzen, zusätzlich zu einem öffentlichen Code-Repository ähnlich Github.com, basierend auf Gitlab. Das wäre sehr zu begrüßen, fördert es doch die Transparenz und ermöglicht eine Nachnutzung in verschiedenen Bundesländern und auf verschiedenen Staatsebenen wie Bund, Ländern und Kommunen. Das könnte in einem föderalen System wie in Deutschland durch eine Nachnutzung von Software große Synergieeffekte haben.
```

Comment:
Diese deutschen Bestrebungen sind seit einigen Tagen für alle öffentlich online, bitte entsprechend in den Text einbauen: https://opencode.de/